src/services/meta_transaction_service.ts

```diff
@@ -162,6 +194,146 @@ export class MetaTransactionService {
         };
         return apiMetaTransactionQuote;
     }
+    public async findTransactionByHashAsync(refHash: string): Promise<TransactionEntity | undefined> {
+        return this._transactionEntityRepository.findOne({
+            where: [{ refHash }, { txHash: refHash }],
+        });
+    }
+    public async validateZeroExTransactionFillAsync(
+        zeroExTransaction: ZeroExTransactionWithoutDomain,
+        signature: string,
+    ): Promise<BigNumber> {
+        // Verify 0x txn won't expire in next 60 seconds
+        // tslint:disable-next-line:custom-no-magic-numbers
+        const sixtySecondsFromNow = new BigNumber(Math.floor(new Date().getTime() / ONE_SECOND_MS) + 60);
+        if (zeroExTransaction.expirationTimeSeconds.lte(sixtySecondsFromNow)) {
+            throw new Error('zeroExTransaction expirationTimeSeconds in less than 60 seconds from now');
+        }
+
+        const decodedArray = await this._devUtils.decodeZeroExTransactionData(zeroExTransaction.data).callAsync();
+        const orders = decodedArray[1];
+
+        // Verify orders don't expire in next 60 seconds
+        orders.forEach(order => {
```

Comment:
suggestion: I don't think this service should concern itself with validating orders, there could still be alternate orders in here which would allow the tx to be successful.